### PR TITLE
BR: fix case sensitive issue in CheckRestoreDBAndTable (#32400)

### LINKED ISSUE #32400: table has not been backup, please ensure you has input a correct table name
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. create table with uppercase letter. `create table T(id int);`
2. use table backup and specify the name with lowercase letter. `tiup br:nightly backup table --db tpcc --table t ...`
3. use table restore and specify the name with lowercase letter.

### 2. What did you expect to see? (Required)
restore successful
### 3. What did you see instead (Required)
restore failed with error `table t has not been backup, please ensure you has input a correct table name`
### 4. What is your TiDB version? (Required)
v5.3.0

### PR DESCRIPTION
**Pull request by MoCuishle28**
Fix case sensitive issue in CheckRestoreDBAndTable (#32400)

### What problem does this PR solve?

Issue Number: close #32400

Problem Summary:
backup and restore both are not case sensitive but the check in restore 
`func CheckRestoreDBAndTable(client *restore.Client, cfg *RestoreConfig) error`
is case sensitive.

### What is changed and how it works?
The changes are as follows:
1. transform `table.Info.Name.O` => `table.Info.Name.L`
2. transform `db.Info.Name.O` => `db.Info.Name.L`
3. add the `GetSysDBONameAndLName` in `br/pkg/utils` to get the original name and lower case name of system DB.
4. in `CheckRestoreDBAndTable` transform `restoreTables` `restoreSchemas` to lower case.
5. add unit test `TestCheckRestoreDBAndTable` in `br/pkg/task`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.
```release-note
This PR fix the case-sensitive issue of checking database names and table names when performing the restore.
```

### CODE REVIEW

**Comment by MoCuishle28** [comment]
\cc @3pointer

File: br/pkg/utils/schema.go [hunk 1]
```diff
@@ -114,3 +114,11 @@ func GetSysDBName(tempDB model.CIStr) (string, bool) {
 	}
 	return tempDB.O[len(temporaryDBNamePrefix):], true
 }
+
+// GetSysDBName get the original name and lower case namne of system DB
+func GetSysDBONameAndLName(tempDB model.CIStr) (string, string, bool) {
```
Comment (YuJuncen) [inline br/pkg/utils/schema.go:119]: Why not just return `(CIStr, bool)`?

**Review by YuJuncen — APPROVED** [review]
LGTM

(BTW, should we consider the flag `--case-sensitive`? If that is enabled, would we meet different error message?)

File: br/pkg/utils/schema.go [hunk 2]
```diff
@@ -114,3 +114,11 @@ func GetSysDBName(tempDB model.CIStr) (string, bool) {
 	}
 	return tempDB.O[len(temporaryDBNamePrefix):], true
 }
+
+// GetSysDBName get the original name and lower case namne of system DB
```
Comment (3pointer) [inline br/pkg/utils/schema.go:118]:
```suggestion
// GetSysDBName get the original name and lower case name of system DB
```

**Review by 3pointer — APPROVED** [review]
LGTM

**Comment by 3pointer** [comment]
> LGTM
> 
> (BTW, should we consider the flag `--case-sensitive`? If that is enabled, would we meet different error message?)

case-sensitive seems fine. since this PR only check the restore DB and restore table commands.

**Review by joccau — COMMENTED** [review]
rest LGTM

**Comment by hawkingrei** [comment]
@MoCuishle28 Please unite your signature for your commits.

**Comment by MoCuishle28** [comment]
> @MoCuishle28 Please unite your signature for your commits.

sorry, when I open the page (https://cla.pingcap.net/pingcap/tidb?pullRequest=34912)  to sign the CLA, the page says 
```
Error
There is no CLA to sign for pingcap/tidb
({"error":"bad gateway: net/http: timeout awaiting response headers"})
```

**Pull request merged**